packages/beacon-node/src/network/processor/extractSlotRootFns.ts

```diff
@@ -0,0 +1,30 @@
+import {SlotRootHex} from "@lodestar/types";
+import {
+  getBlockRootFromAttestationSerialized,
+  getBlockRootFromSignedAggregateAndProofSerialized,
+  getSlotFromAttestationSerialized,
+  getSlotFromSignedAggregateAndProofSerialized,
```

Comment:
@tuyennhv Can you add unit tests for 
- getBlockRootFromAttestationSerialized
- getBlockRootFromSignedAggregateAndProofSerialized,
- getSlotFromAttestationSerialized,
- getSlotFromSignedAggregateAndProofSerialized,

For each: decode an empty buffer, and decode a buffer that ends right after the offset such that this function produces incomplete data. The unit test should ensure that this functions never throw unexpectedly and let's think what to do in that case

Reply:
I added invalid test data to those functions. I think we should let the consumer decide what to do instead of never throw and catch error silently
+ If getting block or slot throws error, we ignore the forkchoice check to handle unknown block root, when deserializing the data it'll throw error and reject
+ if getAttDataBase64 throws error, it acts like not having a cached Attestation Data